be/src/query/query.service.ts

```diff
@@ -1,24 +1,75 @@
-import { Inject, Injectable } from '@nestjs/common';
+import { Inject, Injectable, NotFoundException } from '@nestjs/common';
 import { QueryDto } from './dto/query.dto';
 import { QUERY_DB_ADAPTER } from '../config/query-database/query-db.moudle';
 import { QueryDBAdapter } from '../config/query-database/query-db.adapter';
-import { ResQueryDto } from './dto/res-query.dto';
+import { QueryType } from '../common/enums/query-type.enum';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Shell } from '../shell/shell.entity';
+import { Repository } from 'typeorm';
 
 @Injectable()
 export class QueryService {
   constructor(
     @Inject(QUERY_DB_ADAPTER) private readonly queryDBAdapter: QueryDBAdapter,
+    @InjectRepository(Shell)
+    private shellRepository: Repository<Shell>,
   ) {}
 
-  async execute(userId: string, queryDto: QueryDto) {
-    const connection = await this.queryDBAdapter.createConnection(userId);
+  async execute(sessionId: string, shellId: number, queryDto: QueryDto) {
+    const shell = await this.shellRepository.findOne({ where: { shellId } });
+    if (!shell) {
+      throw new NotFoundException('존재하지 않는 shellId 입니다.');
+    }
+
+    const connection = await this.queryDBAdapter.createConnection(sessionId);
+    const baseUpdateData = {
+      query: queryDto.query,
+      queryType: this.detectQueryType(queryDto.query),
+    };
+
     try {
       const rows = await this.queryDBAdapter.run(connection, queryDto.query);
-      return ResQueryDto.ok(queryDto.query, rows);
+      const slicedRows = rows.length > 100 ? rows.slice(0, 100) : rows;
+
+      await this.updateShell(shellId, shell, {
+        ...baseUpdateData,
+        affectedRows: rows.length,
+        queryStatus: true,
+        resultTable: slicedRows,
+      });
     } catch (e) {
-      return ResQueryDto.fail(e.sqlMessage);
+      await this.updateShell(shellId, shell, {
+        ...baseUpdateData,
+        queryStatus: false,
+        failMessage: e.sqlMessage,
+      });
     } finally {
       this.queryDBAdapter.closeConnection(connection);
     }
+
+    return shell;
+  }
+
+  private async updateShell(
+    shellId: number,
+    shell: Shell,
+    updateData: Partial<Shell>,
+  ) {
+    await this.shellRepository.update({ shellId }, updateData);
+    Object.assign(shell, updateData);
+  }
```

Comment:
쉘 업데이트 서비스에서, 쉘을 매개변수로 받아오는 것보다는 서비스 내부에서 찾아주는게 더 좋을 것 같아요. 쉘 자체를 매개변수로 받아오면 하나의 서비스로서 기능하지 못하는 것 같아요.
```suggestion
  private async updateShell(
    shellId: number,
    updateData: Partial<Shell>,
  ) {
    const shell = await this.shellRepository.findOne({ where: { shellId } });
    await this.shellRepository.update({ shellId }, updateData);
    Object.assign(shell, updateData);
  }
```

Reply:
쉘 update관련은 쉘 서비스에서 따로 만들어서 처리하는 방식으로 바꾸겠습니당